### ISSUE TITLE
RBTree deepcopy issue

### ISSUE THREAD

**Issue by kadartamascsaba** [mozman/bintrees#3]
Hi!

I think, I found a little issue in RBTree deepcopy process.

I used the latest version with BUGFIX.

```
C:\Users\user>pip list
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
bintrees (2.0.6)
pip (9.0.1)
setuptools (28.8.0)
```

I tried to make a deepcopy, but I did not succeed.

```
C:\Users\user>python
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:42:59) [MSC v.1500 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import copy
>>> import bintrees
>>> t1 = bintrees.RBTree()
>>> t2 = copy.deepcopy(t1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\lib\copy.py", line 174, in deepcopy
    y = copier(memo)
  File "C:\Python27\lib\site-packages\bintrees\abctree.py", line 234, in __deepcopy__
    self.foreach(_deepcopy, order=-1)
  File "C:\Python27\lib\site-packages\bintrees\abctree.py", line 658, in foreach
    _traverse(self._root)
  File "C:\Python27\lib\site-packages\bintrees\abctree.py", line 649, in _traverse
    func(node.key, node.value)
AttributeError: 'NoneType' object has no attribute 'key'
>>>
```

Can you help me in the issue?

Thanking you in advance!

**Comment by mozman** [comment]
Bug in foreach is fixed in v2.0.7

**Comment by kadartamascsaba** [comment]
Thank you very much!
